### ISSUE TITLE
Gome2 gather stage issue

### ISSUE THREAD

**Issue by JGulic** [NextGeoss/ckanext-nextgeossharvest#66]
### CKAN Version if known (or site URL)
The Gome2 harvester with the improvements it's deployed on staging: http://147.228.242.207/ 

### Please describe the expected behaviour
Sometimes the new version of the Gome2 harvester doesn't collect the data. During the gather stage you can see that there are datasets that need to be gathered, but the gather stage finishes with: No harvester objects to fetch. Here is the log from the staging server:
```
ckan_1        | 2018-09-19 11:05:32,537 INFO  [provider_logger] gome-2       | 2018-09-19 11:05:32.309089 | 200 | 0.22446s
ckan_1        | 2018-09-19 11:05:32,540 DEBUG [ckanext.nextgeossharvest.lib.gome2_base] GOME2_O3_2018-09-18 has not been harvested before. Attempting to harvest it.
ckan_1        | 2018-09-19 11:05:32,732 INFO  [provider_logger] gome-2       | 2018-09-19 11:05:32.540675 | 200 | 0.188077s
ckan_1        | 2018-09-19 11:05:32,734 DEBUG [ckanext.nextgeossharvest.lib.gome2_base] GOME2_O3_2018-09-19 has not been harvested before. Attempting to harvest it.
ckan_1        | 2018-09-19 11:05:32,911 INFO  [provider_logger] gome-2       | 2018-09-19 11:05:32.735012 | 200 | 0.173032s
ckan_1        | 2018-09-19 11:05:32,915 DEBUG [ckanext.nextgeossharvest.lib.gome2_base] GOME2_NO2_2018-09-16 will not be updated.
ckan_1        | 2018-09-19 11:05:32,917 DEBUG [ckanext.nextgeossharvest.lib.gome2_base] GOME2_NO2_2018-09-17 has not been harvested before. Attempting to harvest it.
ckan_1        | 2018-09-19 11:05:33,117 INFO  [provider_logger] gome-2       | 2018-09-19 11:05:32.917525 | 200 | 0.196534s
ckan_1        | 2018-09-19 11:05:33,120 DEBUG [ckanext.nextgeossharvest.lib.gome2_base] GOME2_NO2_2018-09-18 has not been harvested before. Attempting to harvest it.
ckan_1        | 2018-09-19 11:05:33,308 INFO  [provider_logger] gome-2       | 2018-09-19 11:05:33.120575 | 200 | 0.18425s
ckan_1        | 2018-09-19 11:05:33,311 DEBUG [ckanext.nextgeossharvest.lib.gome2_base] GOME2_NO2_2018-09-19 has not been harvested before. Attempting to harvest it.
ckan_1        | 2018-09-19 11:05:33,485 INFO  [provider_logger] gome-2       | 2018-09-19 11:05:33.311295 | 200 | 0.1692s
ckan_1        | 2018-09-19 11:05:33,488 DEBUG [ckanext.nextgeossharvest.lib.gome2_base] GOME2_SO2_2018-09-16 has not been harvested before. Attempting to harvest it.
ckan_1        | 2018-09-19 11:05:33,700 INFO  [provider_logger] gome-2       | 2018-09-19 11:05:33.488863 | 200 | 0.207479s
ckan_1        | 2018-09-19 11:05:33,702 DEBUG [ckanext.nextgeossharvest.lib.gome2_base] GOME2_SO2_2018-09-17 has not been harvested before. Attempting to harvest it.
ckan_1        | 2018-09-19 11:05:33,894 INFO  [provider_logger] gome-2       | 2018-09-19 11:05:33.703013 | 200 | 0.188099s
ckan_1        | 2018-09-19 11:05:33,897 DEBUG [ckanext.nextgeossharvest.lib.gome2_base] GOME2_SO2_2018-09-18 has not been harvested before. Attempting to harvest it.
ckan_1        | 2018-09-19 11:05:34,077 INFO  [provider_logger] gome-2       | 2018-09-19 11:05:33.897376 | 200 | 0.176712s
ckan_1        | 2018-09-19 11:05:34,080 DEBUG [ckanext.nextgeossharvest.lib.gome2_base] GOME2_SO2_2018-09-19 has not been harvested before. Attempting to harvest it.
ckan_1        | 2018-09-19 11:05:34,247 INFO  [provider_logger] gome-2       | 2018-09-19 11:05:34.080346 | 200 | 0.16365s
ckan_1        | 2018-09-19 11:05:34,250 DEBUG [ckanext.nextgeossharvest.lib.gome2_base] GOME2_SO2mass_2018-09-16 has not been harvested before. Attempting to harvest it.
ckan_1        | 2018-09-19 11:05:34,457 INFO  [provider_logger] gome-2       | 2018-09-19 11:05:34.250584 | 200 | 0.20331s
ckan_1        | 2018-09-19 11:05:34,459 DEBUG [ckanext.nextgeossharvest.lib.gome2_base] GOME2_SO2mass_2018-09-17 has not been harvested before. Attempting to harvest it.
ckan_1        | 2018-09-19 11:05:34,649 INFO  [provider_logger] gome-2       | 2018-09-19 11:05:34.459962 | 200 | 0.185496s
ckan_1        | 2018-09-19 11:05:34,653 DEBUG [ckanext.nextgeossharvest.lib.gome2_base] GOME2_SO2mass_2018-09-18 has not been harvested before. Attempting to harvest it.
ckan_1        | 2018-09-19 11:05:34,845 INFO  [provider_logger] gome-2       | 2018-09-19 11:05:34.653418 | 200 | 0.188456s
ckan_1        | 2018-09-19 11:05:34,849 DEBUG [ckanext.nextgeossharvest.lib.gome2_base] GOME2_SO2mass_2018-09-19 has not been harvested before. Attempting to harvest it.
ckan_1        | 2018-09-19 11:05:35,029 INFO  [provider_logger] gome-2       | 2018-09-19 11:05:34.849445 | 200 | 0.176039s
ckan_1        | 2018-09-19 11:05:35,033 DEBUG [ckanext.nextgeossharvest.lib.gome2_base] GOME2_TropNO2_2018-09-16 will not be updated.
ckan_1        | 2018-09-19 11:05:35,035 DEBUG [ckanext.nextgeossharvest.lib.gome2_base] GOME2_TropNO2_2018-09-17 has not been harvested before. Attempting to harvest it.
ckan_1        | 2018-09-19 11:05:35,233 INFO  [provider_logger] gome-2       | 2018-09-19 11:05:35.035413 | 200 | 0.194241s
ckan_1        | 2018-09-19 11:05:35,236 DEBUG [ckanext.nextgeossharvest.lib.gome2_base] GOME2_TropNO2_2018-09-18 has not been harvested before. Attempting to harvest it.
ckan_1        | 2018-09-19 11:05:35,421 INFO  [provider_logger] gome-2       | 2018-09-19 11:05:35.236774 | 200 | 0.180343s
ckan_1        | 2018-09-19 11:05:35,424 DEBUG [ckanext.nextgeossharvest.lib.gome2_base] GOME2_TropNO2_2018-09-19 has not been harvested before. Attempting to harvest it.
ckan_1        | 2018-09-19 11:05:35,595 INFO  [provider_logger] gome-2       | 2018-09-19 11:05:35.424403 | 200 | 0.167428s
ckan_1        | 2018-09-19 11:05:35,605 INFO  [ckanext.harvest.queue] No harvest objects to fetch
```
The harvester run successfully few times and then this started to happen.
This may cause problems later on when a harvester object has been created, but a dataset not.

### Please describe the actual behaviour
We need to be sure that the gather stage will finish with passing objects, if any, to the fetch stage or to the import stage and the datasets should be imported in the database.

### What steps can be taken to reproduce the issue?
Try to start the GOME2 harvester to run automatically. If this doesn't help you to reproduce the issue, you can use the version that we had, before the improvements with the stop date and you will be able to reproduce this issue locally as well.

**Comment by joaandrade** [comment]
@JGulic can you send me the configuration and the start date you used? I will run the last code delivered for the same configuration.

**Comment by joaandrade** [comment]
@JGulic the harvester was running properly, the problem was related with the logs. Logs were updated. We have tested the harvester since 2018-02-01 and until today 661 datasets were created as expected:

METOP_A_GOME2_O3 - 209
METOP_A_GOME2_NO2 - 217
METOP_A_GOME2_TropNO2 - 210
METOP_A_GOME2_SO2 - 11
METOP_A_GOME2_SO2mass - 14

Some days, we have no GOME-2 products for some collections, however, the server which is queried is able to retrieve an empty file. After that, we have to check the "missing days" query to understand which are the products that should be created in our catalogue (the ones that really exist and not the empty ones). It happens that the logs were printing the message "has not been harvested... Attempt to harvest it" for the empty files which are not supposed to be harvested, and then these empty products were not sent to the fetch.

Now the logs were corrected and for this "ghost datasets" the message is: "(identifier) is missing on the original Data Source so it will not be harvested."

Hope this helps and if you are able to harvest 661 datasets from 2018-02-01 until 2018-09-19 everything is fine.
